themes/qgrid/source/js/menu.js

```diff
@@ -18,23 +18,180 @@ function toggleVisibility(e) {
+								dataFound.innerHTML = foundTag[k] + ' ' + foundTag[k + 1] + ' ...';
+							}
+							else if (foundTag.length > 2 && (k == foundTag.length - 1 || k == foundTag.length - 2)) {
+								dataFound.innerHTML = '... ' + foundTag[k - 1] + ' ' + foundTag[k];
+							}
+							else {
+								dataFound.innerHTML = tag[j];
+							}
+							li[i].getElementsByClassName('title')[0].style.color='grey';
+							li[i].getElementsByClassName('title')[0].style.paddingLeft='8.8%';
+							li[i].getElementsByTagName('data-found')[0].style.display = 'inline';
+							dataFound.setAttribute('href', li[i].getElementsByClassName('title')[0].getAttribute('href'));
+
+							dataFound.style.display = 'inline';
+						}
+						break next;
+					}
+					else {
+
+						if (li[i].getElementsByTagName('data-found')[0]) {
+							li[i].getElementsByTagName('data-found')[0].style.display = 'none';
+							li[i].getElementsByClassName('title')[0].style.font = '';
+							li[i].getElementsByClassName('title')[0].style.paddingLeft = '';
+							li[i].getElementsByClassName('title')[0].style.color= '';
+							li[i].getElementsByClassName('title')[0].innerHTML = li[i].getElementsByClassName('title')[0].innerHTML.replace("/ ", '');
+
+
+
+						}
 
+					}
+				}
+			}
+		}
+	}
+	let foundTitle = [];
 	for (let i = 0; i < li.length; i++) {
-		const a = li[i].getElementsByTagName('a')[0];
-		if (a) {
-			if (a.text.toLowerCase().indexOf(filter) > -1) {
-				li[i].style.display = '';
-			} else {
-				li[i].style.display = 'none';
+		foundTitle[i] = 0;
+		for (let count = 0; count <= li[i].getElementsByTagName('a').length; count++) {
+			const a = li[i].getElementsByTagName('a')[count];
+			if (a) {
+				const index = a.text.toLowerCase().indexOf(search);
+				if (index > -1) {
+					foundTitle[i]++;
+					const symbol = a.text.split('');
+					li[i].style.display = '';
+					for (let j = 0; j < filter.length; j++) {
+						symbol[index + j] = '<font>' + symbol[index + j] + '</font>';
+					}
+					a.innerHTML = symbol.join('');
+					let font = li[i].getElementsByTagName('font');
+					for (let i = 0; i < font.length; i++) {
+						font[i].style.backgroundColor = '#efdf00';
+					}
+				} else {
+					foundTitle[i]--;
+					let font = a.getElementsByTagName('font');
+					for (let j = 0; j < font.length; j++) {
+						font[j].style.backgroundColor = '';
+					}
+					if (-foundTitle[i] == li[i].getElementsByTagName('a').length) {
+						li[i].style.display = 'none';
+					}
+
+
+				}
+			}
+		}
+	}
+	localStorage.setItem('input', location.hash.substr(8));
+}
+
+function searching(e) {
+	e.stopImmediatePropagation();
+	if (location.hash) { localStorage.setItem('input', location.hash.substr(8).replace(/%20/g, ' ')); };
+	const active = document.getElementsByClassName('active-topic')[0];
+	if (active) {
+		const tags = active.dataset.tag;
+		const index = tags.toLowerCase().indexOf(localStorage.getItem('input').replace(/%20/g, ' '));
```

Comment:
You can look how it's done in q-grid itself here: https://github.com/qgrid/ng2/blob/master/src/lib/pipes/highlight.pipe.ts